jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -116,12 +134,212 @@ MlirToJlmConverter::ConvertOperation(
   else if (auto constant = ::mlir::dyn_cast<::mlir::arith::ConstantIntOp>(&mlirOperation))
   {
     auto type = constant.getType();
+    // std::cout << "Casted to ConstantIntOp with t value: " << constant.value() << "\n";
     JLM_ASSERT(type.getTypeID() == ::mlir::IntegerType::getTypeID());
-    auto integerType = ::mlir::cast<::mlir::IntegerType>(type);
+    auto integerType = type.isIntOrFloat() ? type.cast<::mlir::IntegerType>() : nullptr;
 
     return rvsdg::node_output::node(
         rvsdg::create_bitconstant(&rvsdgRegion, integerType.getWidth(), constant.value()));
   }
+
+  // Binary Comparision operations
+  else if (auto ComOp = ::mlir::dyn_cast<::mlir::arith::CmpIOp>(&mlirOperation))
+  {
+    if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::eq)
+    {
+      return rvsdg::node_output::node(rvsdg::biteq_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::ne)
+    {
+      auto st = dynamic_cast<const jlm::rvsdg::bittype *>(&inputs[0]->type());
+      if (!st)
+        throw jlm::util::error("frontend : expected bitstring type for ne operation.");
```

Comment:
Use `JLM_ASSERT()` instead of the exception, please, or handled the  floating point type. However, then it should also be handled in the `eq` case, no?